File: DynamoDB-SDK-Examples/node.js/WorkingWithItems/put-item.js
// This is an example of a simple PutItem using the higher level DocumentClient for Amazon DynamoDB

const { DynamoDBClient } = require("@aws-sdk/client-dynamodb");
const { DynamoDBDocumentClient, PutCommand } = require("@aws-sdk/lib-dynamodb");

async function putItems() {
  const client = new DynamoDBClient({ region: "us-west-2" });
  const ddbDocClient = DynamoDBDocumentClient.from(client);

  return await ddbDocClient.send(
      new PutCommand({
        TableName: "RetailDatabase",
        Item: {
          pk: "jim.Robert@somewhere.com",
          sk: "metadata",
          name: "Jim Roberts",
          first_name: "Jim",
          last_name: "Roberts",
          address: {
            road: "456 Nowhere Lane",
            city: "Langely",
            state: "WA",
            pcode: "98260",
            country: "USA",
          },
          username: "jrob",
        },
      })
  );
}

putItems()
    .then((data) =>
        console.log("PutItem succeeded with HTTP code:", JSON.stringify(data.$metadata.httpStatusCode, null, 2))
    )
    .catch((error) => console.error(JSON.stringify(error, null, 2)));

File: examples/SDK/python/WorkingWithItems/put_item.py
from __future__ import print_function # Python 2/3 compatibility
import boto3, json, decimal

from botocore.exceptions import ClientError

# Helper class to convert a DynamoDB item to JSON.
class DecimalEncoder(json.JSONEncoder):
    def default(self, o):
        if isinstance(o, decimal.Decimal):
            if abs(o) % 1 > 0:
                return float(o)
            else:
                return int(o)
        return super(DecimalEncoder, self).default(o)

dynamodb = boto3.resource('dynamodb', region_name='us-west-2')

table = dynamodb.Table('RetailDatabase')

#Insert new user data into the system
try:
    response = table.put_item(
       Item={
            'pk': "jim.bob@somewhere.com",
            'sk': "metadata",
            'name': "Jim Bob",
            'first_name': "Jim",
            'last_name': "Bob",
            'address': {
                'road': "456 Nowhere Lane",
                'city': "Langely",
                'state': "WA",
                'pcode': "98260",
                'country': "USA"
            },
            'username': "jbob"
        }
    )
except ClientError as e:
    print(e.response['Error']['Message'])
else:
    print("PutItem succeeded:")
    print(json.dumps(response, indent=4, cls=DecimalEncoder))